lisa/operating_system.py

```diff
@@ -332,6 +352,37 @@ def _initialize_package_installation(self) -> None:
         # sub os can override it, but it's optional
         pass
 
+    def _get_package_information(self, package_name: str) -> VersionInfo:
+        raise NotImplementedError()
+
+    def _get_version_info_from_named_regex_match(
+        self, package_name: str, named_matches: re.Match[str]
+    ) -> VersionInfo:
+
+        assert_that(named_matches.group("package_name")).described_as(
+            "While capturing package version info, found name of package was "
+            "unexpected. Version info may be incorrect or regex may need updating."
+        ).is_equal_to(package_name)
+        patch_match = named_matches.group("patch")
+        if patch_match:
```

Comment:
Should it be `not patch_match`? And is it possible to leave the patch as is? The default "0" may be confusing sometime.

Reply:
Yep, sorry I'm committing here to be able to cherry pick to my test branch.
VersionInfo requires an entry for 'patch' since it expects three numbers though. If there's a patch version released it should have the entry, I noticed some packages leave it off when it's 0 (dpdk).

Reply:
ok, we can add a default "0".